extended/src/main/java/io/kubernetes/client/workqueue/DefaultDelayingQueue.java

```diff
@@ -0,0 +1,70 @@
+package io.kubernetes.client.workqueue;
+
+import com.google.common.primitives.Longs;
+
+import java.time.Duration;
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The default delaying queue implementation.
+ */
+public class DefaultDelayingQueue extends DefaultWorkQueue implements DelayingQueue {
+
+    private DelayQueue      delayQueue    = new DelayQueue();
+    private ExecutorService waitingWorker = Executors.newSingleThreadExecutor();
+
+    public DefaultDelayingQueue() {
```

Comment:
It's probably worth making the `ExecutorService` an optional parameter, for testing and other reasons.

Reply:
done.